Src/Sarif.PatternMatcher/SearchSkimmer.cs

```diff
@@ -19,9 +19,13 @@ namespace Microsoft.CodeAnalysis.Sarif.PatternMatcher
     public class SearchSkimmer : Skimmer<AnalyzeContext>
     {
         public const string AssetFingerprint = "AssetFingerprint/v1";
-        public const string GlobalFingerprint = "GlobalFingerprint/v1";
         public const string ValidationFingerprint = "ValidationFingerprint/v1";
         public const string ValidationFingerprintHash = "ValidationFingerprintHash/v1";
+
+        public const string AssetFingerprintV2 = "AssetFingerprint/v2";
+        public const string ValidationFingerprintV2 = "ValidationFingerprint/v2";
+        public const string ValidationFingerprintHashV2 = "ValidationFingerprintHash/v2";
```

Comment:
>ValidationFingerprintHash [](http://example.com/codeflow?start=59&length=25)

Ok, so thinking this through some more. First, our validation hashes aren't necessary to emit, strictly speaking. The reason is that they are simply a SHA256 hash of other data that's already in the log file. Also, the new v2 hash doesn't afford any new value, strictly speaking, i.e., the v1 hash is still perfectly sufficient for its purpose. The reason that we're replacing it is because of encoding problems parsing the unhashed version. But the hashed version works fine.

And so, what we could do here is to either not emit the v2 hash for now (i.e., continue to use the v1). Note that anyone is still free to hash the v2 validation fingerprint.

Alternately, we could go implement that new hash that omits the property names. Now *this* hash is not something that could be computed from other log data and this value *would* provide additional value, in that it would subsequently be resilient to any property name changes. If/when we have this, we could emit is as a v2 value.

Thoughts?

Reply:
I think we can remove the hash-v2 for now and re-think the use of values only to generate the hash as u described above and i described in the other comment :)